File: LeetCode_ProblemSet/Problem1534.cpp
#include <vector>
#include <cmath>

class Solution {
    public:
    int countGoodTriplets(std::vector<int>& arr, int a, int b, int c) {
        int cnt = 0;
        for (int i = 0; i < arr.size(); i++) {
            for (int j = i + 1; j < arr.size(); j++) {
                if (std::abs(arr[i] - arr[j]) > a) continue;

                for (int k = j + 1; k < arr.size(); k++) {
                    if (std::abs(arr[j] - arr[k]) > b) continue;
                    if (std::abs(arr[i] - arr[k]) > c) continue;
                    cnt++;
                }
            }
        }
        return cnt;
    }
};

File: LeetCode_ProblemSet/Rust/src/Problem1534.rs
impl Solution {
    pub fn count_good_triplets(arr: Vec<i32>, a: i32, b: i32, c: i32) -> i32 {
        let mut cnt = 0;
        for i in 0..arr.len() {
            for j in (i + 1)..arr.len() {
                if (arr[i] - arr[j]).abs() > a {
                    continue;
                }
                for k in (j + 1)..arr.len() {
                    if (arr[j] - arr[k]).abs() > b {
                        continue;
                    }
                    if (arr[i] - arr[k]).abs() > c {
                        continue;
                    }
                    cnt += 1;
                }
            }
        }
        return cnt;
    }
}